src/ol/renderer/canvas/canvasmaprenderer.js

```diff
@@ -193,11 +227,15 @@ ol.renderer.canvas.Map.prototype.renderFrame = function(frameState) {
       continue;
     }
     if (layerRenderer.prepareFrame(frameState, layerState)) {
+      if (layer instanceof ol.layer.Vector && layer.getSource().getWrapX()) {
```

Comment:
Sounds a bit arbitrary to me. As if the renderer decided for the user...

Did you consider the simplification where we never wrap the world for vectors drawn in postcompose?